src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -423,6 +419,15 @@ public void updateReplicateStatus(OptimisticLockingFailureException exception,
                                                                 String walletAddress,
                                                                 ReplicateStatusUpdate statusUpdate,
                                                                 UpdateReplicateStatusArgs updateReplicateStatusArgs) {
+        // Synchronization is mandatory there to avoid race conditions.
+        // Lock key should be unique, e.g. `chainTaskId + walletAddress`.
+        return replicatesUpdateLockRunner.getWithLock(
+                chainTaskId + walletAddress,
+                () -> updateReplicateStatusWithoutSync(chainTaskId, walletAddress, statusUpdate, updateReplicateStatusArgs)
+        );
+    }
+
+    private Optional<TaskNotificationType> updateReplicateStatusWithoutSync(String chainTaskId, String walletAddress, ReplicateStatusUpdate statusUpdate, UpdateReplicateStatusArgs updateReplicateStatusArgs) {
```

Comment:
What do you mean by "without sync" here? The method itself is not synchronized?

Reply:
That's it. I wanted to add some clear difference with `updateReplicateStatus`. That may not be the right wording, I'm not sure at all.

Reply:
Something like `updateReplicateStatusNow`? (I'm sure you have something better !)

Reply:
We can keep the name of this method as is, and change the calling method (the retryable one) to `updateReplicateStatusAsync`.

Reply:
@zguesmi > The calling method isn't async, so it doesn't work :/
The idea is to tell in the method name that it could be subject to race conditions. But I don't find any short and clear ways to do so.

Reply:
Technically it is not async indeed, but practically it is running the update asynchronously.

Reply:
I don't get it. There's no asynchronously update there. Everything is synchronous in `updateReplicateStatus`, `updateReplicateStatusWithoutSync` and `ContextualLockRunner`.

Reply:
@mcornaton I guess you don't want to rename L418 to
`public Optional<TaskNotificationType> updateReplicateStatusThreadSafely`
to keep a short name from the caller part?

Reply:
Yes, that's my concern. But I still don't have any better solution :thinking:

Reply:
The day we signed on to work with java, there was a hidden statement, "iAgreeToUsingVeryLongLongNamesInJavaDespiteThemBeingVeryLong" :smile:

Reply:
Joking apart, I don't find the name `updateReplicateStatusThreadSafely` to be that long.
The other option would be to rename `updateReplicateStatusWithoutSync` to `updateReplicateStatusWithoutThreadSafety` (for example) with a comment explaining why and which method to use in which case.

Reply:
I prefer the second option as it should be used a single time ; it won't add lots of characters overall.